pkg/reconcilermanager/controllers/permissions.go

```diff
@@ -25,11 +29,43 @@ func rolereference(name, kind string) rbacv1.RoleRef {
 	}
 }
 
-// subject returns and initialized Subject with kind, name and namespace.
-func subject(name, namespace, kind string) rbacv1.Subject {
+// newSubject returns and initialized Subject with kind, name and namespace.
+func newSubject(name, namespace, kind string) rbacv1.Subject {
 	return rbacv1.Subject{
 		Kind:      kind,
 		Name:      name,
 		Namespace: namespace,
 	}
 }
+
+// findSubjectIndex returns the index of the expected subject in the list, or -1.
+func findSubjectIndex(list []rbacv1.Subject, expected rbacv1.Subject) int {
+	for i, found := range list {
+		if found == expected {
+			return i
+		}
+	}
+	return -1
+}
+
+// addSubject adds a subject to the subject list.
+// Subjects are sorted, if modified.
+func addSubject(subjects []rbacv1.Subject, subject rbacv1.Subject) []rbacv1.Subject {
+	if i := findSubjectIndex(subjects, subject); i < 0 {
+		subjects = append(subjects, subject)
+		if len(subjects) > 0 {
+			sort.SliceStable(subjects, func(i, j int) bool {
+				return subjects[i].Name < subjects[j].Name
+			})
+		}
+	}
+	return subjects
+}
+
+// removeSubject removes the subject from the subject list.
+func removeSubject(subjects []rbacv1.Subject, subject rbacv1.Subject) []rbacv1.Subject {
+	if i := findSubjectIndex(subjects, subject); i >= 0 {
+		subjects = append(subjects[:i], subjects[i+1:]...)
```

Comment:
Does this error if the element is the last in the slice (from the `i+1` index)?

Reply:
unfortunately, yes... I'll have to fix that.

Reply:
on that note, it wouldn't hurt to have some unit tests for these helper functions

Reply:
fixed and added unit tests for all the new subject list functions. Found another bug with sorting and fixed it.